packages/vats/src/proposals/upgrade-mintHolder-proposal.js

```diff
@@ -0,0 +1,63 @@
+import { makeTracer } from '@agoric/internal';
+import { E } from '@endo/far';
+
+const trace = makeTracer('upgrade mintHolder', true);
+
+export const upgradeMintHolder = async (
+  {
+    consume: {
+      contractKits: contractKitsP,
+      instancePrivateArgs: instancePrivateArgsP,
+    },
+  },
+  options,
+) => {
+  const { contractRef, label } = options.options;
+  assert(contractRef.bundleID, 'mintHolder bundleID not found');
+  assert(label, 'mintHolder bank asset label not found');
+
+  trace(`Start ${label} mintHolder contract upgrade`);
+
+  const [contractKits, instancePrivateArgs] = await Promise.all([
+    contractKitsP,
+    instancePrivateArgsP,
+  ]);
+
+  const mintHolderKit = Array.from(contractKits.values()).filter(
+    kit => kit.label && kit.label === label,
+  );
+  trace('mintHolderKit: ', mintHolderKit);
+
+  const { publicFacet, adminFacet, instance } = mintHolderKit[0];
+
+  /**
+   * Ensure that publicFacet holds an issuer by calling makeEmptyPurse, the
+   * core-eval will throw if otherwise.
+   */
+  await E(publicFacet).makeEmptyPurse();
```

Comment:
Can we verify that the method is present without calling it? try `__getMethodNames__()`.

Reply:
I was not able to find an example of getMethodNames() being used on Agoric code, and while trying your suggestion it triggers the following error:

```
target has no method __getMethodNames__ , has [ 'burn', 'getAllegedName', 'getAmountOf', 'getAssetKind', 'getBrand', 'getDisplayInfo', 'isLive', 'makeEmptyPurse' ]
```

For that reason, I am invoking the makeEmptyPurse method itself

Reply:
calling `makeEmptyPurse()` seems better than bothering with `__getMethodNames__()`.